registry/storage/driver/s3/s3.go

```diff
@@ -314,6 +321,9 @@ func (d *driver) WriteStream(path string, offset int64, reader io.Reader) (total
 				}
 			}
 		}
+
+		d.putbuf(buf)
```

Comment:
This defer is run at the end of the function. I guess these can be outside this defer.